examples/zipkin/src/main/scala/com/example/http4s/zipkin/SimpleRelayServerApp.scala

```diff
@@ -0,0 +1,69 @@
+package com.example.http4s.zipkin
+
+import org.http4s.client.Client
+import org.http4s.client.blaze.PooledHttp1Client
+import org.http4s.dsl._
+import org.http4s.server.blaze._
+import org.http4s.server.{Server, ServerApp}
+import org.http4s.zipkin.client.{ClientRequirements, ZipkinClient}
+import org.http4s.zipkin.core.algebras.{Clock, Randomness}
+import org.http4s.zipkin.core.interpreters.collector.Http
+import org.http4s.zipkin.core.Endpoint
+import org.http4s.zipkin.middleware.server.ZipkinService
+import org.http4s.zipkin.server.{ServerRequirements, ZipkinServer}
+import org.http4s.{HttpService, Uri}
+
+import scalaz.concurrent.Task
+
+class SimpleRelayServerApp(
+  getConfig: Task[Config],
+  serviceDiscovery: String => Task[Uri],
+  randomness: Randomness, clock: Clock
+) extends ServerApp {
+
+  val http1Client = PooledHttp1Client()
+  val collector = Http(http1Client)
+
+  val instrument: Client => ZipkinClient =
+    ZipkinClient(collector, randomness, clock)
+
+  val zipkinClient: ZipkinClient = instrument(http1Client)
+
+  def serviceWithZipkinClient(nextServiceName: ServiceName)(
+    serverRequirements: ServerRequirements
+  ): HttpService = {
+
```

Comment:
What about:
`val client = zipkinClient.run(ClientRequirements(serverRequirements.serverIds, nextServiceName)` here?  Then the client call in the for-comprehension gets easier to read and looks more like what you're really trying to do and less like Zipkin ceremony.